JobsPortal/src/main/java/com/ps/js/controller/JobsController.java

```diff
@@ -0,0 +1,363 @@
+	 * 
+	 * @return {@link ResponseEntity}
+	 */
+
+	@GetMapping("/find-all-job")
+	public ResponseEntity<List<JobDetails>> findAllJob() {
+		// Fetch all the jobs available
+		List<JobDetails> listJobDetails = jobDetailsService.findAllJob();
+		if (listJobDetails == null)
+			throw new JobDetailsNotFoundException(ErrorMessages.NO_JOB_DETAILS_FOUND_EXCEPTION.toString());
+
+		return new ResponseEntity<List<JobDetails>>(listJobDetails, HttpStatus.OK);
+	}
+
+	/**
+	 * For create a new jobDetails and return jobDetails with generated jobId
+	 * 
+	 * @param jobDetailsPayload
+	 * @return {@link ResponseEntity}
+	 */
+	@PostMapping("/open-job")
+	public ResponseEntity<JobDetailsResponsePayload> openJob(@Valid @RequestBody JobDetailsPayload jobDetailsPayload) {
+		Set<JobLocation> locations = new HashSet<JobLocation>();
+		Set<Skill> primarySkills = new HashSet<Skill>();
+		Set<Skill> secondrySkills = new HashSet<Skill>();
+		Set<RolesAndResponsebility> rolesAndResponsebilities = new HashSet<RolesAndResponsebility>();
+		JobDetails jobDetails = new JobDetails();
+		JobDetailsResponsePayload jobDetailsResponsePayload = null;
+		// Convert job details to job details payload using jobDetailsMapper service
+		jobDetails = jobDetailsMapper.jobDetailsPayloadToJobDetailsMapper(jobDetailsPayload, jobDetails);
+
+		System.out.println("jobDetails :: " + jobDetails);
+
+		// Fetching the location and adding to collection
+		for (JobLocation location : jobDetails.getJobLocation()) {
+			locations.add(jobLocationServices.fetchJobLocationById(location.getLocationId()).get());
+		}
+
+		// Fetching the primary skills and added to collection
+		for (Skill skill : jobDetails.getPrimarySkill()) {
+			primarySkills.add(skillServices.fetchSkillById(skill.getSkillId()).get());
+		}
+
+		// Fetching the secondary skills and added to collection
+		for (Skill skill : jobDetails.getSecondrySkill()) {
+			secondrySkills.add(skillServices.fetchSkillById(skill.getSkillId()).get());
+		}
+
+		// Fetching the roles and responsebility and added to collection
+		for (RolesAndResponsebility responsebility : jobDetails.getRolesAndResponsebilities()) {
+			rolesAndResponsebilities
+					.add(rolesAndResponsebilityService.fetchRolesAndResponsebilityById(responsebility.getId()).get());
+		}
+		if (locations.isEmpty())
+			throw new LocationsNotFoundException();
+		else if (primarySkills.isEmpty())
+			throw new SkillNotFoundException(ErrorMessages.PRIMARY_SKILL_NOT__FOUND_EXCEPTION.toString());
+		else if (secondrySkills.isEmpty())
+			throw new SkillNotFoundException(ErrorMessages.SECONDRY_SKILL_NOT__FOUND_EXCEPTION.toString());
+
+		// Invoking jobDetailsService to add a new jobDetails
+		jobDetailsResponsePayload = jobDetailsService.createJob(jobDetails);
+
+		if (jobDetailsResponsePayload != null) {
+			for (ResponsePayload payload : jobDetailsResponsePayload.getJobLocation()) {
+				payload.setUrl("jobs/find-location-by-id/" + payload.getId());
```

Comment:
Please post a response payload you see in the browser or Postman and also trace and confirm, is this what you wanted to send as a response. I mean about the setUrl function you used.

Reply:
we can have a discussion on this comment when available!